File: org.kevoree.modeling.microframework/src/main/java/org/kevoree/modeling/api/Extrapolations.java
package org.kevoree.modeling.api;

import org.kevoree.modeling.api.strategy.DiscreteExtrapolationStrategy;
import org.kevoree.modeling.api.strategy.ExtrapolationStrategy;
import org.kevoree.modeling.api.strategy.LinearRegressionExtrapolationStrategy;
import org.kevoree.modeling.api.strategy.PolynomialExtrapolationStrategy;

/**
 * Created by duke on 10/28/14.
 */
public enum Extrapolations {

    DISCRETE(new DiscreteExtrapolationStrategy()), LINEAR_REGRESSION(new LinearRegressionExtrapolationStrategy()), POLYNOMIAL(new PolynomialExtrapolationStrategy());

    private Extrapolations(ExtrapolationStrategy wrappedStrategy) {
        this.extrapolationStrategy = wrappedStrategy;
    }

    private ExtrapolationStrategy extrapolationStrategy;

    public ExtrapolationStrategy strategy() {
        return extrapolationStrategy;
    }

}

File: org.kevoree.modeling.microframework.typescript/src/main/ts/org/kevoree/modeling/api/Extrapolations.ts
///<reference path="strategy/DiscreteExtrapolationStrategy.ts"/>
///<reference path="strategy/ExtrapolationStrategy.ts"/>
///<reference path="strategy/LinearRegressionExtrapolationStrategy.ts"/>
///<reference path="strategy/PolynomialExtrapolationStrategy.ts"/>

class Extrapolations {

  public static DISCRETE: Extrapolations = new Extrapolations(new DiscreteExtrapolationStrategy());
  public static LINEAR_REGRESSION: Extrapolations = new Extrapolations(new LinearRegressionExtrapolationStrategy());
  public static POLYNOMIAL: Extrapolations = new Extrapolations(new PolynomialExtrapolationStrategy());
  private extrapolationStrategy: ExtrapolationStrategy = null;

  constructor(wrappedStrategy: ExtrapolationStrategy) {
    this.extrapolationStrategy = wrappedStrategy;
  }

  public strategy(): ExtrapolationStrategy {
    return this.extrapolationStrategy;
  }

  public equals(other: AccessMode): boolean {
        return this == other;
    }

}

